cardano-cli/src/Cardano/CLI/Shelley/Run/Genesis.hs

```diff
@@ -828,74 +838,74 @@ createDelegateKeys :: FilePath -> Word -> ExceptT ShelleyGenesisCmdError IO ()
 createDelegateKeys dir index = do
   liftIO $ createDirectoryIfMissing False dir
   runGenesisKeyGenDelegate
-        (VerificationKeyFile $ dir </> "delegate" ++ strIndex ++ ".vkey")
-        coldSK
-        opCertCtr
+        (File @OfVerificationKey $ dir </> "delegate" ++ strIndex ++ ".vkey")
+        (onlyOut coldSK)
+        (onlyOut opCertCtr)
   runGenesisKeyGenDelegateVRF
-        (VerificationKeyFile $ dir </> "delegate" ++ strIndex ++ ".vrf.vkey")
-        (SigningKeyFile $ dir </> "delegate" ++ strIndex ++ ".vrf.skey")
+        (File @OfVerificationKey $ dir </> "delegate" ++ strIndex ++ ".vrf.vkey")
+        (File @OfSigningKey $ dir </> "delegate" ++ strIndex ++ ".vrf.skey")
   firstExceptT ShelleyGenesisCmdNodeCmdError $ do
     runNodeKeyGenKES
-        kesVK
-        (SigningKeyFile $ dir </> "delegate" ++ strIndex ++ ".kes.skey")
+        (onlyOut kesVK)
+        (File @OfSigningKey $ dir </> "delegate" ++ strIndex ++ ".kes.skey")
     runNodeIssueOpCert
-        (VerificationKeyFilePath kesVK)
-        coldSK
+        (VerificationKeyFilePath (onlyIn kesVK))
+        (onlyIn coldSK)
```

Comment:
To use an `InOut` file for an `In` operation, we can cast it.  This works on all file types regardless of content type.